### PR TITLE
Editorial: Cite Image Resource

### LINKED ISSUE #854: Use the newly defined ImageResource
The `ImageResource` definition moved to its own [spec](https://w3c.github.io/image-resource/). The Manifest's definition includes some extra manifest-specific fields. So instead of defining its own `ImageResource` it could re-use parts of the new `ImageResource` by defining the following dictionary:

```webidl
dictionary ManifestImageResource : ImageResource {
  USVString platform;
  USVString purpose;
}
```

### PR DESCRIPTION
**Pull request by christianliebel**
Closes #854

This change (choose one):

* [ ] Breaks existing normative behavior (please add label "breaking")
* [ ] Adds new normative requirements
* [ ] Adds new normative recommendations or optional items
* [x] Makes only editorial changes (only changes informative sections, or
  changes normative sections without changing behavior)
* [ ] Is a "chore" (metadata, formatting, fixing warnings, etc).

Commit message:

Cite the Image Resource spec.

***
<a href="https://pr-preview.s3.amazonaws.com/christianliebel/manifest/pull/895.html" title="Last updated on Jun 9, 2020, 1:47 AM UTC (5026d09)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/manifest/895/a58d2cc...christianliebel:5026d09.html" title="Last updated on Jun 9, 2020, 1:47 AM UTC (5026d09)">Diff</a>

### CODE REVIEW

**Comment by rayankans** [comment]
Looks good to me as well. I'm wondering if `ShortcutItem` should use `ManifestImageResource` or just `ImageResource`. Any thoughts on that @aarongustafson ?

**Comment by aarongustafson** [comment]
> Looks good to me as well. I'm wondering if `ShortcutItem` should use `ManifestImageResource` or just `ImageResource`. Any thoughts on that @aarongustafson ?

Could we see `purpose` being used in the shortcut context?

**Comment by kenchris** [comment]
The new monochrome purpose could be preferred by some Android launchers I think

**Comment by NotWoods** [comment]
The maskable purpose could also be preferred for full-bleed shortcut images such as those for contacts:
https://developer.android.com/guide/topics/ui/shortcuts

File: index.html [hunk 1]
```diff
@@ -1139,11 +1139,11 @@ <h3>
           <var>manifest</var>["<a>categories</a>"].
           </li>
           <li>Set <var>manifest</var>["<a>icons</a>"] to the result of running
-          <a>processing ImageResource members</a> given
+          <a>processing ManifestImageResource members</a> given
```
Comment (marcoscaceres) [inline index.html:1142]:
```suggestion
          <a>processing `ManifestImageResource` members</a> given
```

File: index.html [hunk 2]
```diff
@@ -1139,11 +1139,11 @@ <h3>
...
           <var>manifest</var>["<a>icons</a>"] and <var>manifest URL</var>.
           </li>
           <li>Set <var>manifest</var>["<a>screenshots</a>"] to the result of
-          running <a>processing ImageResource members</a> given
+          running <a>processing ManifestImageResource members</a> given
```
Comment (marcoscaceres) [inline index.html:1146]:
```suggestion
          running <a>processing `ManifestImageResource` members</a> given
```

File: index.html [hunk 3]
```diff
@@ -1513,12 +1513,12 @@ <h3>
         </h3>
         <p>
           The <dfn>icons</dfn> member is an <a>array</a> of
-          <a>ImageResource</a>s that can serve as iconic representations of the
-          web application in various contexts. For example, they can be used to
-          represent the web application amongst a list of other applications,
-          or to integrate the web application with an <abbr title=
-          "operating system">OS</abbr>'s task switcher and/or system
-          preferences.
+          <a>ManifestImageResource</a>s that can serve as iconic
```
Comment (marcoscaceres) [inline index.html:1516]:
```suggestion
          {{ManifestImageResource}}s that can serve as iconic
```

File: index.html [hunk 4]
```diff
@@ -1528,7 +1528,7 @@ <h3>
           examination, to in fact be inappropriate (e.g. because its content
           type is unsupported), then the user agent MUST try the
           next-most-appropriate icon as determined by examining the
-          <a>ImageResource</a>'s members.
+          <a>ManifestImageResource</a>'s members.
```
Comment (marcoscaceres) [inline index.html:1531]:
```suggestion
          {{ManifestImageResource}}'s members.
```

File: index.html [hunk 5]
```diff
@@ -1891,8 +1891,8 @@ <h3>
         </h3>
         <p>
           The <dfn>screenshots</dfn> member is an <a>array</a> of
-          <a>ImageResource</a>s, representing the web application in common
-          usage scenarios.
+          <a>ManifestImageResource</a>s, representing the web application in
```
Comment (marcoscaceres) [inline index.html:1894]:
```suggestion
          {{ManifestImageResource}}s, representing the web application in
```

File: index.html [hunk 6]
```diff
@@ -2062,82 +2062,31 @@ <h3>
...
-        web application, suitable to use in various contexts depending on the
-        semantics of the member that is using the object (e.g., an icon that is
-        part of an application menu, etc.). For an image resource, this
-        specification provides developers with a means of specifying the
-        dimensions, and <a data-cite="mimesniff">MIME type</a> of an image
-        (i.e., a "responsive image" solution [[RESPIMG-USECASES]]). A user
-        agent can use these values to select an image that is best suited to
-        display on the end-user's device or most closely matches the end-user's
-        preferences.
+        Each <a>ManifestImageResource</a> represents an image that is used as
```
Comment (marcoscaceres) [inline index.html:2076]:
```suggestion
        Each {{ManifestImageResource}} represents an image that is used as
```

File: index.html [hunk 7]
```diff
@@ -2439,158 +2400,46 @@ <h2 class="icon-title">
...
-                  <li>Set <var>image</var>["platform"] to
-                  <var>entry</var>["platform"].
-                  </li>
-                  <li>[=list/append=] <var>image</var> to
-                  <var>imageResources</var>
-                  </li>
-                </ol>
+              <li>Let |image| be the result of running [=process an image
+              resource from JSON|processing an image resource from JSON=] given
+              |entry| and |manifest URL|.
```
Comment (marcoscaceres) [inline index.html:2422]:
```suggestion
              <li>Let |image| be the result of running [=process an image resource from JSON=] given
              |entry| and |manifest URL|.
```

**Pull request merged**